rplugin/python3/deoplete/sources/tabnine.py

```diff
@@ -82,15 +82,12 @@ def gather_candidates(self, context):
         candidates = []
         self.debug(repr(response))
         for result in response['results']:
-            candidate = {}
-            word = result['new_prefix']
-            suffix = result['old_suffix']
-            if suffix and word.endswith(suffix):
-                candidate['word'] = word[:len(word)-len(suffix)]
-                candidate['word'] += result['new_suffix']
-                candidate['abbr'] = word
-            else:
-                candidate['word'] = word
+            candidate = {'word': result['new_prefix']}
+            if result['old_suffix'] or result['new_suffix']:
+                candidate['user_data'] = json.dumps({
```

Comment:
What's user_data?

Reply:
`:help complete-items`

```
						*complete-items*
Each list item can either be a string or a Dictionary.  When it is a string it
is used as the completion.  When it is a Dictionary it can contain these
items:
	word		the text that will be inserted, mandatory
	abbr		abbreviation of "word"; when not empty it is used in
			the menu instead of "word"
	menu		extra text for the popup menu, displayed after "word"
			or "abbr"
	info		more information about the item, can be displayed in a
			preview window
	kind		single letter indicating the type of completion
	icase		when non-zero case is to be ignored when comparing
			items to be equal; when omitted zero is used, thus
			items that only differ in case are added
	equal		when non-zero, always treat this item to be equal when
			comparing. Which means, "equal=1" disables filtering
			of this item.
	dup		when non-zero this match will be added even when an
			item with the same word is already present.
	empty		when non-zero this match will be added even when it is
			an empty string
	user_data 	custom data which is associated with the item and
			available in |v:completed_item|
```

Reply:
Huh so it's not documented how deoplete handles user_data. Would be a good idea to add that documentation somewhere.